### PR TITLE
MGMT-8757: convert proxy package

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
Convert proxy package to be a typed interface, with mock file.
Update the unit test for this package and the package
that are using proxy package

### CODE REVIEW

File: pkg/proxy/proxy_test.go [hunk 1]
```diff
@@ -33,7 +37,7 @@ var _ = Describe("Setup", func() {
 
 		uo := unstructured.Unstructured{Object: m}
 
-		err = proxy.Setup(&uo)
+		err = proxyStruct.Setup(&uo)
```
Comment (ybettan) [inline pkg/proxy/proxy_test.go:40]: Can we make `Setup()` a private method if it is not part of the API?

Comment (yevgeny-shnaidman) [reply pkg/proxy/proxy_test.go:40]: it is part of the API, it is defined in the interface

Comment (ybettan) [reply pkg/proxy/proxy_test.go:40]: Sorry, I missed that.

**Pull request merged**